### PR TITLE
feat: added pinecone vectorstore

### PR DESCRIPTION
**Pull request by brenkao**
* Added PineconeVectorStore
* Updated BaseVectorStore to support both Pinecone and Chroma

### CODE REVIEW

File: mirascope/pinecone/types.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+from typing import Any, Literal, Optional
+
+from pinecone.config import Config
+from pinecone.core.client.api.manage_indexes_api import ManageIndexesApi
+from pydantic import BaseModel, ConfigDict
+
+from ..rag.types import BaseVectorStoreParams
+
+
+class ServerlessSpec(BaseModel):
```
Comment (willbakst) [inline mirascope/pinecone/types.py:11]: none of these classes have docstrings? or API docs

File: mirascope/pinecone/vectorstores.py [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+    ] = PineconeServerlessParams(cloud="aws", region="us-east-1")
+    client_settings: ClassVar[PineconeSettings] = PineconeSettings()
+
+    def retrieve(self, text: str, **kwargs: Any) -> PineconeQueryResult:
+        """Queries the vectorstore for closest match"""
+        embed = self.embedder.embed
+        if self.vectorstore_params.weave is not None and not isinstance(
+            self.chunker, weave.Op
+        ):
+            embed = self.vectorstore_params.weave(self.embedder.embed)
```
Comment (willbakst) [inline mirascope/pinecone/vectorstores.py:70]: you're missing a pragma no cover here

File: mirascope/pinecone/vectorstores.py [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+                )  # pragma: no cover
+            documents = chunk(text)
+        else:
+            documents = text
+        inputs = [document.text for document in documents]
+        embed = self.embedder.embed
+        if self.vectorstore_params.weave is not None and not isinstance(
+            self.chunker, weave.Op
+        ):
+            embed = self.vectorstore_params.weave(self.embedder.embed)
```
Comment (willbakst) [inline mirascope/pinecone/vectorstores.py:123]: same here

File: mirascope/rag/types.py [hunk 4]
```diff
@@ -1,8 +1,36 @@
...
+
+class BaseEmbeddingResponse(BaseModel, Generic[ResponseT], ABC):
+    """A base abstract interface for LLM embedding responses.
+
+    Attributes:
+        response: The original response from whichever model response this wraps.
+    """
+
+    response: ResponseT
+    start_time: float  # The start time of the completion in ms
```
Comment (willbakst) [inline mirascope/rag/types.py:18]: comment here seems copied. Do we actually set start/end time?

File: pyproject.toml [hunk 5]
```diff
@@ -31,15 +31,17 @@ anthropic = { version = ">=0.23.1,<1.0.0", optional = true }
 mistralai = { version = ">=0.1.6,<1.0.0", optional = true }
 groq = { version = ">=0.4.2,<1.0.0", optional = true }
 cohere = { version = "^5.2.5", optional = true }
+pinecone-client = {version = "^3.2.2", optional = true}
```
Comment (willbakst) [inline pyproject.toml:34]: nit: add spaces between brackets and internals

File: tests/chroma/conftest.py [hunk 6]
```diff
@@ -1,8 +1,22 @@
...
+
+
 import pytest
 
 from mirascope.chroma.types import ChromaSettings
 from mirascope.chroma.vectorstores import ChromaVectorStore
+from mirascope.rag import BaseEmbedder
+
+
+class TestEmbedder(BaseEmbedder):
```
Comment (willbakst) [inline tests/chroma/conftest.py:11]: can you name it this? I often run into issues naming a class Test* in a test file

File: mirascope/rag/types.py (hunk 4 above)
Comment (brenkao) [reply mirascope/rag/types.py:18]: yes for openai embeddings

File: tests/chroma/conftest.py (hunk 6 above)
Comment (brenkao) [reply tests/chroma/conftest.py:11]: it didnt complain but i will change it

File: mirascope/rag/types.py (hunk 4 above)
Comment (willbakst) [reply mirascope/rag/types.py:18]: ok, maybe update to say "the start time of embedding the text in ms"?

**Review by willbakst — APPROVED** [review]

**Pull request merged**